projects/dotcms-js/src/lib/core/core-web.service.ts

```diff
@@ -135,77 +149,104 @@ export class CoreWebService {
-            this.httpErrosSubjects[httpErrorCode] = new Subject();
+    private handleHttpError(response: HttpErrorResponse): void {
+        if (!this.httpErrosSubjects[response.status]) {
+            this.httpErrosSubjects[response.status] = new Subject();
         }
 
-        return this.httpErrosSubjects[httpErrorCode].asObservable();
+        this.httpErrosSubjects[response.status].next(response);
     }
 
-    private getRequestOpts(options: RequestOptionsArgs): Request {
-        const headers: Headers = this._apiRoot.getDefaultRequestHeaders();
+    private getRequestOpts<T>(options: RequestOptionsArgs): HttpRequest<T> {
+        const optionsArgs: RequestOptionsParams = {
+            headers: new HttpHeaders(),
+            params: new HttpParams()
+        };
+
+        optionsArgs.headers = this._apiRoot.getDefaultRequestHeaders();
         const tempHeaders = options.headers
-            ? options.headers
+            ? options.headers.toJSON()
             : { 'Content-Type': 'application/json' };
 
         Object.keys(tempHeaders).forEach((key) => {
-            headers.set(key, tempHeaders[key]);
+            optionsArgs.headers = optionsArgs.headers.set(key, tempHeaders[key]);
         });
 
-        // https://github.com/angular/angular/issues/10612#issuecomment-238712920
-        options.body =
+        const body =
             options.body && typeof options.body !== 'string'
                 ? JSON.stringify(options.body)
-                : options.body
-                    ? options.body
-                    : '';
-
-        options.headers = headers;
+                : options.body;
 
         if (options.url.indexOf('://') === -1) {
-
-            options.url = options.url.startsWith('/api') ?
-                `${this._apiRoot.baseUrl}${options.url.substr(1)}`
+            options.url = options.url.startsWith('/api')
+                ? `${this._apiRoot.baseUrl}${options.url.substr(1)}`
                 : `${this._apiRoot.baseUrl}api/${options.url}`;
         }
 
         if (this.browserUtil.isIE11()) {
-            options = options || {};
-            options.search = options.search || new URLSearchParams();
-            const currentTime = new Date().getTime();
-            (<URLSearchParams>options.search).set('timestamp', String(currentTime));
+            optionsArgs.params = optionsArgs.params.set('timestamp', String(new Date().getTime()));
         }
 
-        return new Request(<any>options);
-    }
+        if (options.params) {
+            optionsArgs.params = this.setHttpParams(
+                <URLSearchParams>options.params,
+                optionsArgs.params
+            );
+        }
 
-    private handleHttpError(response): void {
-        if (!this.httpErrosSubjects[response.status]) {
-            this.httpErrosSubjects[response.status] = new Subject();
+        if (options.search) {
+            optionsArgs.params = this.setHttpParams(
+                <URLSearchParams>options.search,
```

Comment:
Why you need the casting? should not be typed as this already?

Reply:
in `RequestOptionsArgs.params` and `RequestOptionsArgs.search` accepts both `string|URLSearchParams`
https://docs.w3cub.com/angular~7/api/http/requestoptionsargs/

but when passes on this function https://github.com/dotCMS/core-web/blob/edb8514321f58c0979a2e055f3f30bbdc43e1f97/projects/dotcms-js/src/lib/core/core-web.service.ts#L255

we need to specify that only `URLSearchParams` is passed, otherwise it complains